auth-service/src/main/java/com/kioke/auth/service/AuthService.java

```diff
@@ -0,0 +1,95 @@
+package com.kioke.auth.service;
+
+import com.kioke.auth.constant.KiokeServices;
+import com.kioke.auth.exception.ServiceFailedException;
+import com.kioke.auth.exception.ServiceNotFoundException;
+import com.kioke.auth.exception.UserAlreadyExistsException;
+import com.kioke.auth.exception.UserDoesNotExistException;
+import com.kioke.auth.model.User;
+import com.kioke.auth.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+
+@Service
+@Slf4j
+public class AuthService implements UserDetailsService {
```

Comment:
The UserService already exists, wouldn't it be better if the UserService implements the UserDetailsService interface?